src/components/velocity.rs

```diff
@@ -16,21 +16,18 @@ pub struct Velocity {
 pub(crate) struct VelocityPlugin;
 
 impl VelocityPlugin {
-    fn update_velocity(
-        mut velocity: Query<(&mut Velocity, &GlobalTransform)>,
-        time: Res<FixedTime>,
-    ) {
+    fn update_velocity(mut velocity: Query<(&mut Velocity, &GlobalTransform)>, time: Res<Time>) {
         velocity.iter_mut().for_each(|(mut velocity, transform)| {
             let current_position = transform.translation();
             let delta_position = current_position - velocity.last_position;
-            velocity.current_velocity = delta_position / time.period.as_secs_f32();
+            velocity.current_velocity = delta_position / time.delta().as_secs_f32();
             velocity.last_position = current_position;
         })
     }
 }
 
 impl Plugin for VelocityPlugin {
     fn build(&self, app: &mut App) {
-        app.add_systems(Update, Self::update_velocity);
```

Comment:
Why this change? Doesn't this reintroduce the glitchy doppler effect?

https://github.com/Salzian/bevy_fmod/commit/8af8e8f666dcb006bc53260179359641ecf9ee7b#diff-819c727ab5d1508bb4a39ccd5fad25915fe88923d71f1d3c13dccb6131f839daL35

Reply:
Hmm I thought it didn't but I just checked again with a sine wave and it does :( guess it's more subtle than I remembered.

Anyways, I think we were mis-using the time API previously, so with the 0.12 migration I couldn't figure out how to replicate the same behavior. Because I think we were actually using a fixed time in the Update schedule, which does not have a fixed time.

I'll need more time to look into this

Reply:
Tried something, feels like a clunky solution but it's way better. Not a 100% as good as old but you can barely hear it with a pure sine wave

Reply:
What it sounds like https://youtu.be/ckAXzCvEJAY?feature=shared

Reply:
> What it sounds like https://youtu.be/ckAXzCvEJAY?feature=shared

I also tried this example with bevy_framepace and it made it a lot worse. This is a hint in what is going wrong but I haven't had the time yet to find out exactly what it is